kratos/python_scripts/file_handler_utility.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Other imports
+import os
+
+class FileHandlerUtility(object):
```

Comment:
What about `AsciiFileWriter`. Utility feels like a bit of a noise word to me and the class seems much more specialized than its name suggests.

Reply:
Besides that, I feel like this module is not that _general_ and am a bit concerned about releasing it into the wild. It appears to have at least two responsibilities:

1. Control the flush frequency.
2. Manage restarts.

For point 1, I suggest using the buffering parameter in `open` instead of flush frequency (see https://lyceum-allotments.github.io/2017/03/python-and-pipes-part-4-on-the-buffers/). This will simplify a lot.

If you eliminate the flush frequency, then for point 2 I think you really only need the `AddToExistingOutputFile`. This function make assumptions about the file format which are not enforced by the `FileHandlerUtility`. E.g., header lines begin with `#`, each line starts with a time value. This is most certainly the case for point_output_process but not in general. Maybe this function could be put in the same module as point_output_process and we can eliminate file_handler_utility.py?

Reply:
Thx for the suggestions

Regarding the name I am very open to improvements.

Flushing seems also better with your suggestion, I will change that

Regarding the restarts: I would like to later also use this for other processes that print data. Therefore I put it to a separate class.

Would it be ok for you if the name indicates the format? Sth like `TimeBasedAsciiFileWriter`? Still not the best name but I think it conveys the idea

Reply:
Ok, in that case, it might make sense to make `TimeBasedAsciiFileWriter` a process and initialize the ascii file in the `ExecuteInitialize` so the restart details are implemented there.